projects/knora-ui/src/lib/viewer/values/date-value/date-value.component.html

```diff
@@ -1,19 +1,24 @@
-<span [formGroup]="form">
-    <mat-form-field class="large-field child-value-component" floatLabel="never">
-      <kui-date-input #dateInput [formControlName]="'dateValue'" class="value" placeholder="Date value" [readonly]="mode === 'read'" [errorStateMatcher]="matcher"></kui-date-input>
-    </mat-form-field>
-    <mat-form-field class="large-field value-component-comment" *ngIf="shouldShowComment">
-        <textarea matInput
-                  cdkTextareaAutosize
-                  #autosize="cdkTextareaAutosize"
-                  cdkAutosizeMinRows="1"
-                  cdkAutosizeMaxRows="6"
-                  [formControlName]="'comment'"
-                  class="comment"
-                  placeholder="Comment"
-                  type="text"
-                  [readonly]="mode === 'read'"
-                  spellcheck="false">
-        </textarea>
-    </mat-form-field>
-</span>
+<div *ngIf="mode === 'read'; else showForm" class="read-mode-view">
+    <div *ngIf="valueFormControl.value.end; else date">
```

Comment:
Whenever the `FormControl`'s value is expected to be an object (complex values) and you want to access a member, you should use the ["safe navigation operator" ?](https://angular.io/guide/template-syntax#the-safe-navigation-operator----and-null-property-paths): `valueFormControl.value?.end`.

In read mode, the value should always be a `KnoraDate` and also the resetting of the `FormControl`s via `ngOnChanges` should happen at the right time (namely before the [view](https://angular.io/api/core/OnChanges#ngonchanges) is checked), but it's better to be safe and prevent type errors.

Reply:
good to know! done in 37865b7